03-Bash/fix_owner/fix_owner.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+FORCE=0
+
+function help()
+{
+    echo -e "Usage: fix_owner <OPTION>\nFor each user in a system with a UID greater than or equal to 1000 and\nlower than or equal to 60000, checks and fixes UID and GID of items\nin the home directory (recursively).\n\nOption:\n\t-f, --force\t\tfix UID and GID without user confirmation\n\t-h, --help\t\tprint this message and exit"
+}
+
+# $1 -- "UID" or "GID"
+# $2 -- file name
+function need_fix_prompt()
+{
+    ANS_ACQUIRED=0
+    while [ $ANS_ACQUIRED -eq 0 ]; do
+        read -p "Do you want to change $1 for $2 [y/n]? " choice </dev/tty
```

Comment:
Is it required to use /dev/tty?

Reply:
Is there any other way to specify where from to read input in nested loops? Do you have a hint? A function with this loop is used in another while loop.

Reply:
The other possible solution is to dup file descriptor for /dev/stdin and use it in nested loop.
Use line
exec 3<&0
before redirection

read -p"prompt:" choice <&3
...
and close created file descriptor in the end 
exec 3<&-

Reply:
In my opinion, using an extra descriptor for this purpose is overhead. Just like using temporary files instead of pipes.